pkg/util/k8s.go

```diff
@@ -30,6 +30,47 @@ func PodStartedBefore(p, q *corev1.Pod) bool {
 	return p.Status.StartTime.Before(q.Status.StartTime)
 }
 
+// PodPreferredAZIfPresent returns the desired availability zone of the Pod, if it has one
+func PodPreferredAZIfPresent(pod *corev1.Pod) string {
+	if pod.Spec.Affinity == nil || pod.Spec.Affinity.NodeAffinity == nil {
+		return ""
+	}
+
+	affinity := pod.Spec.Affinity.NodeAffinity
+
+	azForTerm := func(term corev1.NodeSelectorTerm) string {
```

Comment:
Can this function be standalone? If so, I'd prefer that way, it's more natural.

Reply:
Fair enough, done